File: Frontend/src/events/converters/requisite.converter.ts
import type { RequisiteDTO } from '../dto/journals/quests/requisite.dto';
import type { Requisite } from '../../models/Journal';

export class RequisiteConverter {
    public static convert(requisiteDto: RequisiteDTO): Requisite {
        return {
            id: requisiteDto.id,
            isDone: requisiteDto.value !== undefined && requisiteDto.value !== 0
        };
    }
}

File: Backend/Events/Converters/Journal/RequisiteConverter.cs
using Backend.Domain.Models.Journals.Quests;
using Backend.Events.DTO.Journal;

namespace Backend.Events.Converters.Journal;

public static class RequisiteConverter
{
    public static RequisiteDTO ToDTO(Requisite requisite) => new()
    {
        Id = requisite.Id,
        Value = requisite.Value
    };
}
